### DISCUSSION TITLE
Generated service method not accepting request with metadata

### DISCUSSION THREAD

**Discussion by gitmalong** [hyperium/tonic#1247] [Q&A]
Hi!

I need to add metadata to my request but when wrapping the original request struct into a new one with metadata the types of my auto-generated service method mismatch. 

```rust
let mut client = ExampleClient::connect("http://[::1]:50051").await?;
let (sender, receiver) = tokio::sync::mpsc::unbounded_channel::<ExampleRequest>();
let receiver_stream = tokio_stream::wrappers::UnboundedReceiverStream::new(receiver);

let sub_req = ExampleRequest {
    // ..
};

// Re-create/wrap request to add metadata
let sub_req: Request<ExampleRequest> =
    Request::from_parts(Default::default(), Default::default(), sub_req);

//  ^^^^^^^ expected struct `ExampleRequest`, found struct `tonic::Request`
sender.send(sub_req);

client.subscribe(receiver_stream);

```

Do I need to build the client in a different way?
```rust
// build.rs
fn main() {
    tonic_build::compile_protos("proto/example.proto")
        .unwrap_or_else(|e| panic!("Failed to compile protos {:?}", e));
}
```
Does the metadata definition has to be added to the proto files?

Appreciate any help or ideas! Thanks!

**Comment by gitmalong** [comment]
I found this example to add metadata for all channel requests: https://github.com/hyperium/tonic/blob/df8dd896daa8e722a1cebc61cd8e03aca6ecb64c/examples/src/authentication/client.rs

Still wondering how that would work out for individual ones?

**Comment by LucioFranco** [comment]
metadata is only sent in the main request/response objects returned, for a streaming request you can't send metadata during the stream. Only before and after.
